File: backend/GameApp.Application/Services/GameCreateService.cs
using GameApp.Application.UseCases;
using GameApp.Domain.Entities;
using GameApp.Domain.Repositories;
using GameApp.Domain.ValueObjects.Characters;
using GameApp.Domain.ValueObjects.Scenes;

namespace GameApp.Application.Services;

public class GameCreateService : GameCreateUseCase
{
    private readonly IGameRepository _repo;

    public GameCreateService(IGameRepository repo) => _repo = repo;

    public async Task<Game?> CreateGameAsync(Character character, int numberScenesToFinish, NothingHappensScene finalScene)
    {
        var game = new Game(character, numberScenesToFinish, finalScene);
        return await _repo.SaveAsync(game);
    }
}

File: frontend/gameapp-frontend-js/src/application/services/game-services/game-create-service.js
import { IGameCreateUseCase } from "../../usecases/game-use-cases/game-create-use-case";
import { IGameRepository } from "../../repositories/igame-repository";
import { Game } from "../../../domain/entities/game";

export class GameCreateService extends IGameCreateUseCase {
    constructor(gameRepository) {
        super();
        if (!(gameRepository instanceof IGameRepository)) throw new TypeError("gameRepository must be an instance of IGameRepository");
        this.gameRepository = gameRepository;
    }

    async createGame(character, numberScenesToFinish, finalScene, listCurrentScenes, listCurrentUserActions) {
        const game = new Game(character, numberScenesToFinish, finalScene, listCurrentScenes, listCurrentUserActions);
        const gameSaved = await this.gameRepository.save(game);
        return gameSaved;
    }
}